process-migration-service/src/main/java/org/kie/processmigration/rest/MigrationResource.java

```diff
@@ -0,0 +1,131 @@
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+
+import org.kie.processmigration.model.Credentials;
+import org.kie.processmigration.model.Execution.ExecutionType;
+import org.kie.processmigration.model.Migration;
+import org.kie.processmigration.model.MigrationDefinition;
+import org.kie.processmigration.model.MigrationReport;
+import org.kie.processmigration.model.exceptions.ReScheduleException;
+import org.kie.processmigration.service.CredentialsProviderFactory;
+import org.kie.processmigration.service.MigrationService;
+
+@Path("/migrations")
+@Produces(MediaType.APPLICATION_JSON)
+@ApplicationScoped
+public class MigrationResource {
+
+    @Inject
+    private MigrationService migrationService;
+
+    @GET
+    @Consumes({MediaType.APPLICATION_JSON})
+    public Response findAll() {
+        return Response.ok(migrationService.findAll()).build();
+    }
+
+    @GET
+    @Path("/{id}")
+    public Response get(@PathParam("id") Long id) {
+        Migration migration = migrationService.get(id);
+        if (migration == null) {
+            return getMigrationNotFound(id);
+        }
+        return Response.ok(migration).build();
+    }
+
+    @GET
+    @Path("/{id}/results")
+    public Response getResults(@PathParam("id") Long id) {
+        List<MigrationReport> results = migrationService.getResults(id);
+        if (results == null) {
+            return getMigrationNotFound(id);
+        }
+        return Response.ok(results).build();
+    }
+
+    @POST
+    @Consumes({MediaType.APPLICATION_JSON})
+    public Response submit(@Context HttpHeaders headers, MigrationDefinition migration) {
+        Credentials credentials = CredentialsProviderFactory.getCredentials(headers.getHeaderString(HttpHeaders.AUTHORIZATION));
+        if (credentials == null) {
+            return Response.status(Status.UNAUTHORIZED).header(HttpHeaders.WWW_AUTHENTICATE, "Basic").build();
+        }
+        if (ExecutionType.ASYNC.equals(migration.getExecution().getType())) {
+            return Response.accepted(migrationService.submit(migration, credentials)).build();
+        } else {
+            return Response.ok(migrationService.submit(migration, credentials)).build();
+        }
+    }
+
+    @PUT
+    @Path("/{id}")
+    @Consumes({MediaType.APPLICATION_JSON})
+    public Response update(@Context HttpHeaders headers, @PathParam("id") Long id, MigrationDefinition migrationDefinition) {
+        Credentials credentials = CredentialsProviderFactory.getCredentials(headers.getHeaderString(HttpHeaders.AUTHORIZATION));
+        if (credentials == null) {
+            return Response.status(Status.UNAUTHORIZED).header(HttpHeaders.WWW_AUTHENTICATE, "Basic").build();
+        }
+        try {
+            Migration migration = migrationService.update(id, migrationDefinition, credentials);
+            if (migration == null) {
+                return getMigrationNotFound(id);
```

Comment:
why not throwing this migration not found directly from the service?

Reply:
Indeed, working on it